beacon-chain/utils/shuffle_test.go

```diff
@@ -0,0 +1,33 @@
+package utils
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/prysmaticlabs/prysm/beacon-chain/params"
+)
+
+func TestFaultyShuffle(t *testing.T) {
+	if _, err := Shuffle(common.Hash{'a'}, params.MaxValidators+1); err == nil {
+		t.Error("Shuffle should have failed when validator count exceeds MaxValidators")
+	}
+}
+
+func TestShuffle(t *testing.T) {
+	hash1 := common.BytesToHash([]byte{'a', 'b', 'c', 'd', 'e', 'f', 'g', 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'a', 'b', 'c', 'd', 'e', 'f', 'g'})
+	hash2 := common.BytesToHash([]byte{'1', '2', '3', '4', '5', '6', '7', '1', '2', '3', '4', '5', '6', '7', '1', '2', '3', '4', '5', '6', '7', '1', '2', '3', '4', '5', '6', '7', '1', '2', '3', '4', '5', '6', '7'})
+
+	list1, err := Shuffle(hash1, 100)
+	if err != nil {
+		t.Errorf("Shuffle failed with: %v", err)
+	}
+
+	list2, err := Shuffle(hash2, 100)
+	if err != nil {
+		t.Errorf("Shuffle failed with: %v", err)
+	}
+	if reflect.DeepEqual(list1, list2) {
```

Comment:
What is this actually testing?

Does shuffle guarantee that the output will never be the same order as the input? 
Random shuffle could return the same output as the input if it is truly random. 

Also, list1 and list2 consist of completely different inputs so they would never equal the same even if Shuffle did nothing.

Reply:
It basically test the following algorithm in the spec which implemented by other sharding team: 
https://notes.ethereum.org/SCIg8AH5SA-O4C1G1LYZHQ?view#Helper-functions

it's testing the 2 lists using different seed are going to have different index orders. I haven't figured out a better way to test this algorithm. Let me know if you have something in mind...

Reply:
Oh, I misunderstood what shuffle does. I thought it shuffled the elements in the list, but it actually returns the indices in shuffled order.

What do you think about calling this `ShuffleIndices` for clarity?